include/artifact-uploader.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2017 Prism Skylabs
+ */
+#ifndef CONNECT_SDK_ARTIFACT_UPLOADER_H
+#define CONNECT_SDK_ARTIFACT_UPLOADER_H
+
+#include "domain-types.h"
+#include "boost/shared_ptr.hpp"
+#include "public-util.h"
+
+namespace prism
+{
+namespace connect
+{
+
+class Client;
+// Usage example
+// namespace prc = prism::connect;
+// void configCallback(prc::Client& client)
+// {
+//     client.setLogFlags(prc::Client::LOG_INPUT | prc::Client::LOG_INPUT_JSON);
+//     client.setConnectionTimeoutMs(5000);
+//     client.setLowSpeed(5);
+// }
+//
+// prc::ArtifactUploader::Configuration uploaderConfig("<apiRoot>", "<apiToken>",
+//      "camOne", 32);
+// prc::ArtifactUploader uploader;
+// prc::Status status = uploader.init(uploaderConfig, configCallback);
```

Comment:
Yes. Separating construction and initialization have some benefits. Constructor is usually non-throwing. So one shouldn't care, if something fails in constructor, how inform about it. Also, one can have object defined in one place and initialized in another place. Also this allows separating fast (non-blocking) code into constructor and code, that may potentially block into init().
I will see, what happens, if we try to use uninitialized object and what can be done to prevent/mitigate the risk.